pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -1770,6 +1770,21 @@ type GatewaySpec struct {
 	// +nullable
 	// +optional
 	RgwCommandFlags map[string]string `json:"rgwCommandFlags,omitempty"`
+
+	// RgwReadAffinity defines the read affinity policy to optimize the read requests for the RGW clients
+	// Note: Only supported for Ceph Tentacle (v20)
+	// +optional
+	RgwReadAffinity *RgwReadAffinity `json:"rgwReadAffinity,omitempty"`
+}
+
+type RgwReadAffinity struct {
+	// Type defines the RGW ReadAffinity type
+	// localize: read from the nearest OSD based on crush location of the RGW client
+	// balance: picks a random OSD from the PG's active set
+	// default: read from the primary OSD
+	// +kubebuilder:validation:Enum=localize;balance;default
+	// +required
+	Type string `json:"type"`
```

Comment:
@travisn Added the `RgwReadAffinity` as an object. We might need to add more details here in the future. One could the `failureDomain`.  Users might want to set a different failureDomain other than we they have set in the ObjectStore pool spec.

Reply:
done.